### PR TITLE
Update to the latest Lumino

### PR DESCRIPTION
**Pull request by jtpio**
## References

Extract the Lumino update from https://github.com/jupyterlab/jupyterlab/pull/16794 to another PR, to avoid noise in https://github.com/jupyterlab/jupyterlab/pull/16794 and fix potential issues with this update not necessarily related to https://github.com/jupyterlab/jupyterlab/pull/16794.

Closes https://github.com/jupyterlab/jupyterlab/pull/16674

## Code changes

- [x] Update `@lumino` packages
- [x] Fix accessing the app `_plugins` in the UI tests 
- [x] Fix `overflow: hidden` CSS rule for menus

## User-facing changes

None?

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by jtpio** [comment]
Some of the UI tests failures look more like fixes than real failures:

[lumino-update-fix1.webm](https://github.com/user-attachments/assets/dfb8bf52-9af0-4362-a753-283223f8ef1e)

[lumino-update-fix2.webm](https://github.com/user-attachments/assets/17d59e58-e43e-40a2-b7fa-ec189faf1909)

**Comment by jtpio** [comment]
CI now passing (`examples` failure tracked in https://github.com/jupyterlab/jupyterlab/issues/16772).

**Comment by jtpio** [comment]
Adding to `4.3.0` as it would be great to have to unblock https://github.com/jupyterlab/jupyterlab/pull/16794 and other streams of work.

**Comment by krassowski** [comment]
What do you think about releasing another beta before merging this one? I know it can be annoying as it can cause merge conflicts, but my reasoning is that this will allow to narrow down any regressions by installing different versions?

**Comment by jtpio** [comment]
Yeah it would be great to have this Lumino update out soon so it can be tested with extensions and downstream in Notebook 7.

But then that would mean releasing now, merging this PR, and making another right after (to not wait too long), which can also be tedious.

**Comment by krassowski** [comment]
When I reviewed https://github.com/jupyterlab/lumino/pull/715 I noted that there is a regression with sidebar resizing behaviour, was this addressed? I do not see the changes mentioned in https://github.com/jupyterlab/lumino/pull/715#issuecomment-2271681238 in the diff but maybe it was addressed differently?

> But then that would mean releasing now, merging this PR, and making another right after (to not wait too long), which can also be tedious.

Ok, I will do that - I will take care of resolving conflicts here.

**Comment by jtpio** [comment]
> When I reviewed [jupyterlab/lumino#715](https://github.com/jupyterlab/lumino/pull/715) I noted that there is a regression with sidebar resizing behaviour, was this addressed? I do not see the changes mentioned in [jupyterlab/lumino#715 (comment)](https://github.com/jupyterlab/lumino/pull/715#issuecomment-2271681238) in the diff but maybe it was addressed differently?

Ah maybe this is not covered by this PR then, would need to double check.

The menu overflow issues were caught by the UI tests and should now be fixed though.

**Comment by jtpio** [comment]
Just checked from this PR and there does seem be an issue when resizing the sidebar:

[jupyterlab-lumino-sidebar-resize-issue.webm](https://github.com/user-attachments/assets/a04f1c56-2ce0-4722-8013-375daffbc9f3)

**Comment by krassowski** [comment]
@jtpio rebased to resolve conflicts after release - do you want to take it from here to fix the sidebar styling?

**Comment by krassowski** [comment]
Should we also add an entry about this in the extension migration guide, ideally with instructions how to restore old behaviour in extensions by adding `overflow: hidden`?

**Comment by jtpio** [comment]
Thanks for the suggestion :+1: 

Added in https://github.com/jupyterlab/jupyterlab/pull/16804/commits/39fd0227885328c0b16815003cc6e90722abb5be.

File: docs/source/extension/extension_migration.rst
Comment (krassowski) [inline docs/source/extension/extension_migration.rst:37]:
```suggestion
or (preferably) scope it to the relevant widgets.
```

**Review by krassowski — APPROVED** [review]
LGTM, just tiny suggestion on docs - I think we can continue testing this as beta3 :)

**Comment by jtpio** [comment]
Thanks @krassowski for the review :+1: 

Addressed the last point in https://github.com/jupyterlab/jupyterlab/pull/16804/commits/80304be6ddca35a8a2277e358f07d0758e512fec.

Waiting for CI to complete to get this one in.

**Comment by jtpio** [comment]
The failing `check_links` does not seem related, and should hopefully fix itself soon:

image

**Pull request merged**
